pkg/scheduler/scheduler.go

```diff
@@ -368,6 +370,37 @@ var defaultQueueingHintFn = func(_ klog.Logger, _ *v1.Pod, _, _ interface{}) (fr
 	return framework.Queue, nil
 }
 
+// defaultQueueingHintFnPodUpdate is a QueueingHintFn for pod related events.
+// which is used when the feature gate for QueueingHint is disabled.
+// For Add or Delete events, it always returns Queue.
+// For Update events, it filters out some update events for the efficient requeueing.
+// That doesn't break any behaviour before QueueingHint because actually that has been done since before QueueingHint was implemented.
+var defaultQueueingHintFnPodUpdate = func(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
```

Comment:
https://github.com/kubernetes/kubernetes/pull/122234#issuecomment-1863063666
This comment raises some concerns here:
When I implement this function, my plan is that we can remove this once queueHint hits GA cause after that, all in-tree plugins will have queueHint covered.
But my question is: Do we force a out-of-tree plugin  to implement QueueHint?
Seems currently, we only encourage them to, and in this case, I think many out-of-tree plugins **will not** implement QueueHint. If this pr got merged and a plugin didn't implement queueHint for `podUpdate` event, sth bad will happen https://github.com/kubernetes/kubernetes/pull/122234/files#r1428802717.
So if we just encourage plugin writers to implement QueueHint but not force them to, should we leave this function forever to let scheduler work even when some out-of-tree plugins didn't implement queueHint for `podUpdate`?

Reply:
Do you mean remove this function and use the old `isPodUpdate` like the implementation in new commit?

Then we'll have to implement queueHint for `interpodaffinity` at the same time.
Also there will be problems when users enable schedulingHint feature gate , and their plugin registered podUpdate event but they didn't implement queuehint for podUpdate event.